setup.cfg

```diff
@@ -23,39 +23,29 @@ classifiers =
 [options]
 include_package_data = True
 install_requires =
-    bcrypt==3.2.2
-    click==8.1.3
-    click-option-group==0.5.3
-    elasticsearch==8.3.3
-    fastapi==0.79.1
-    gunicorn==20.1.0
-    ; We temporary pin `h11` to avoid pip downloading the latest version to solve a
-    ; dependency conflict caused by `httpx` which requires httpcore>=0.15.0,<0.16.0 and
-    ; `httpcore` depends on h11>=0.11,<0.13.
-    ; See: https://github.com/encode/httpx/issues/2244
-    h11==0.12.0
-    httpx==0.23.0
-    Jinja2==3.1.2
+    ; By default, we only consider core dependencies required to use Ralph as a
+    ; library (mostly models).
     langcodes==3.3.0
-    ovh==1.0.0
     pydantic[dotenv,email]==1.9.2
-    pymongo[srv]==4.2.0
-    pyparsing==3.0.9
-    python-keystoneclient==5.0.0
-    python-swiftclient==4.0.1
-    requests==2.28.1
     rfc3987==1.3.8
-    sentry_sdk==1.9.5
-    uvicorn[standard]==0.18.2
-    watchgod==0.8.2
-    websockets==10.3
 package_dir =
     =src
 packages = find:
 zip_safe = True
 python_requires = >= 3.9
 
 [options.extras_require]
+backends =
+    elasticsearch==8.3.3
+    ovh==1.0.0
+    pymongo[srv]==4.2.0
+    python-keystoneclient==5.0.0
+    python-swiftclient==4.0.1
+    requests==2.28.1
+    websockets==10.3
```

Comment:
There might be another pattern more adapted for this to avoid installing all backend libraries: entry points.
This is what pylint is doing for example where you add functionalities by installing separate backages. This also allows extension by external projects...

Reply:
If I understand well, what you describe is a dynamic plugin system and I love this idea! I think it's far beyond the scope of this PR, but I'll declare an issue for this. Thanks for this suggestion ❤️

Reply:
So in this PR will you split in several install options?

Reply:
What you are suggesting is to provide one extra-option per backend?

```
# Install only the Elasticsearch backend
$ pip install ralph-malph[backend-es]

# Install Mongo and Swift backends, plus the cli
$ pip install ralph-malph[backend-mongo,backend-swift,cli]
```

If yes, we can do that. Installing the full package, will be a bit verbose, but we can.

Reply:
Up to you to decide. My understanding was that it would always be one of the backends and not several backends at the same time (except maybe for developpers?).

Reply:
I've fixed it in https://github.com/openfun/ralph/pull/218/commits/918206271e7a45849632657bdb1bc6478455b124